scmrepo/git/backend/dulwich/__init__.py

```diff
@@ -344,8 +344,9 @@ def is_tracked(self, path: str) -> bool:
         return False
 
     def is_dirty(self, untracked_files: bool = False) -> bool:
-        staged, unstaged, untracked = self.status()
-        return bool(staged or unstaged or (untracked_files and untracked))
+        kwargs = {} if untracked_files else {"untracked_files": "no"}
+        staged, unstaged, untracked = self.status(**kwargs)  # type: ignore
```

Comment:
Bit confused by mypy here. Found this: https://github.com/python/mypy/issues/11583

Can be replicated with this:

```python
def f(b: bool = True, **kwargs: str) -> None:
    pass

f(**{'x': 'x'})```

Reply:
I think mypy should pass if you explicitly type kwargs as 
```
kwargs: Dict[str, Any] = {} if ...
```